application/frontend/controllers/UserController.php

```diff
@@ -23,14 +55,14 @@ public function actionCreate(): User
         return $user;
     }
 
-    public function actionUpdatePassword(string $employeeId): User
+    public function actionUpdatePassword($employeeId)
     {
-        $user = User::findOne([
-            'employee_id' => $employeeId
-        ]);
+        $user = User::findOne(['employee_id' => $employeeId]);
 
         if ($user === null) {
-            throw new NotFoundHttpException();
+            Yii::$app->response->statusCode = 204;
```

Comment:
no problem, at this time it feels like six of one and half-a-dozen of another but let's see if that proves out over time.  It's an easy change either way and we'll have confidence to make such a change because there will be so many tests behind it all.